tool/dartpad_picker/lib/dartpad_picker.dart

```diff
@@ -51,7 +51,12 @@ class DartPadPicker {
       ..src = iFrameSrc(theme: 'dark', mode: 'dart');
     iFrameHost.children.add(_iFrameElement);
     window.addEventListener('message', (dynamic e) {
-      if (e.data['type'] == 'ready') {
+      // Don't handle events from other iframe elements
+      if (e.data != null &&
+          e.data is Map &&
+          e.data.containsKey('type') &&
+          e.data['type'] is String &&
+          e.data['type'] == 'ready') {
```

Comment:
Hmm, are you saying that there are situations under which `e.data['type'] == 'ready'` is true and yet the new if condition is false? Or is it that the original condition just missing a `?` (as in `e?.data['type'] == 'ready'`), because otherwise it was throwing an exception?

Reply:
The issue was that the `type` attribute isn't set on every event. The YouTube iframe element sends events to the parent window, which don't have the 'type' attribute This adds a few more checks to make sure we are handling the right event.

It might be better to send a 'dartpad-ready' event instead so that we are more certain that the event is coming from the DartPad embed.

Reply:
If the `type` attribute is missing then the value of `e.data['type']` should be null, according to the Dart [Map [] operator semantics](https://api.dartlang.org/stable/2.6.1/dart-core/Map/operator_get.html). And if the value is null, it can't equal `'ready'`.

Is the issue that, in the compiled JS code, `e.data` is a "native" JS value that isn't then adhering to Dart semantics? Thanks for helping me understand.

(Btw, the `e.data != null` term is unnecessary since `e.data is Map` implies that `e.data` isn't null in Dart.)